posts/2021-02-19-configure-trustedHeaderOrigin-21002.adoc

```diff
@@ -0,0 +1,171 @@
+
+The same list as a server.xml example:
+```    
+<httpDispatcher trustedHeaderOrigin="*" trustedSensitiveHeaderOrigin="localhost, 127.0.0.1, 192.168.*.*, 0:0:0:0:0:ffff:*:*, *.ibm.com"/>
+```
+
+For more information about these properties, see link:{url-prefix}docs/21.0.0.2/reference/config/httpDispatcher.html#[Open Liberty docs on httpDispatcher]
+
+
+[#bugs]
+== Significant bugs fixed in this release
+
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A21002+label%3A%22release+bug%22[full list of bugs fixed in 21.0.0.2].
+
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/15433[System WABs may come online with the web container after server reports started]
++
+When a WAB is included in a Liberty feature (also called a system WAB) it is possible that the server started message and the TCP ports will be opened and listening before the WAB has been fully provisioned with the web container. This is different behaviour to web applications WARs where the server started and TCP ports are delayed until all the applications have started or a timeout has occurred on server start.
+
+This could lead to 404 errors early after the server started status has returned until the WABs come online with the web container. The server should delay the TCP ports listening and the server started event for system WABs similar to the way it waits for applications to start.
+
+This has been fixed by delaying the server started message and TCP port listening until the system WABs are ready to serve, or a timeout has occurred. This has been done by introducing a new service type `ServerReadyStatus`, the feature manager will call this service before registering `ServerStarted`. Implementations of `ServerReadyStatus` have the check method called to allow them to block while waiting for the server to be ready before proceeding to register `ServerStarted`.
+
+This allows the `WABInstaller` to check on the status of the system WABs to make sure the have completed their provisioning to the web container. 
+
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/15698[FeatureUtility not parsing liberty custom environment variables]
++
+When the entry is `/${shared.config.dir}/server.xml”` and the environment variable is `WLP_USER_DIR=/test/` in the `test/config/server.xml` file the user recieves an error stating that "it couldn't find file at /user/test/wlp/usr/shared/config/server.xml" where it should have been looking for `test/config/server.xml`.
+
+This bug causes include tags to not be prased and replaces the environment variables found link:https://www.ibm.com/support/knowledgecenter/en/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/twlp_admin_customvars.html[here] 
```

Comment:
```suggestion
This bug causes include tags to not be parsed and replaces the environment variables found link:https://openliberty.io/docs/21.0.0.1/reference/config/server-configuration-overview.html#server-env[here] 
```
Check that new link with the developer

Reply:
Also, not sure if the whole sentence quite makes sense?? It almost reads like the bug causes environment variables to be replaced .... maybe there should be a _not_ in there somewhere??

Reply:
changed to : "This bug caused include tags to not be parsed correctly by not replacing the environment variables found "